great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -1191,11 +1187,15 @@ def create_temporary_table(self, table_name, custom_sql, schema_name=None):
                 table_name=table_name, custom_sql=custom_sql
             )
         elif self.sql_engine_dialect.name.lower() == "snowflake":
-            logger.info("Creating transient table %s" % table_name)
+            if not self.generated_table_name:
```

Comment:
Totally up to you on style, but you could do these 4 lines with a ternary assignment:
`table_type = "TEMPORARY" if self.generated_table_name else "TRANSIENT"`

Reply:
Great suggestion - thanks!